File: SpringMVC_API/src/main/java/paul/spring/mvc/practice/tasks/springdata/TaskEntity.java
package paul.spring.mvc.practice.tasks.springdata;

import paul.spring.mvc.practice.tasks.TaskStatus;

import javax.persistence.*;
import java.util.UUID;

import static java.util.Objects.requireNonNull;

@Entity
@Table(name = "tasks")
public class TaskEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID taskId;
    private String details;
    private TaskStatus taskStatus;

    TaskEntity(){}

    TaskEntity(
            final String details,
            final TaskStatus taskStatus
    ) {
        this(null, details, taskStatus);
    }

    TaskEntity(
            final UUID taskId,
            final String details,
            final TaskStatus taskStatus
    ) {
        this.taskId = taskId;
        this.details = requireNonNull(details);
        this.taskStatus = requireNonNull(taskStatus);
    }

    public UUID getTaskId() {
        return taskId;
    }

    public String getDetails() {
        return details;
    }

    public TaskStatus getTaskStatus() {
        return taskStatus;
    }
}

File: SpringWebflux_Kotlin/todos/src/main/kotlin/com/paul/kotlin/todos/tasks/springdata/TaskEntity.kt
package com.paul.kotlin.todos.tasks.springdata

import com.paul.kotlin.todos.tasks.TaskStatus
import org.springframework.data.annotation.Id
import org.springframework.data.domain.Persistable
import org.springframework.data.relational.core.mapping.Table
import java.util.*
import javax.annotation.processing.Generated
import javax.persistence.*

@Table("TASKS")
data class TaskEntity(
    @Id
    val id: Long?,
    val taskUUID: UUID,
    val details: String,
    val taskStatus: TaskStatus,
)
